application/src/main/java/org/thingsboard/server/service/edge/rpc/EdgeGrpcSession.java

```diff
@@ -774,7 +775,7 @@ private ConnectResponseMsg processConnect(ConnectRequestMsg request) {
             try {
                 if (edge.getSecret().equals(request.getEdgeSecret())) {
                     sessionOpenListener.accept(edge.getId(), this);
-                    this.edgeVersion = request.getEdgeVersion();
+                    this.edgeVersion = processGetAndSaveEdgeVersion(request.getEdgeVersion());
```

Comment:
This looks over complicated. 
I suggest next code:
this.edgeVersion = request.getEdgeVersion();
saveEdgeVersion(request.getEdgeVersion());